app/src/main/java/org/odk/odknotifications/DatabaseCommunicator/ServerDatabaseCommunicator.java

```diff
@@ -0,0 +1,301 @@
+import org.odk.odknotifications.Model.Notification;
+import org.odk.odknotifications.Model.Response;
+import org.opendatakit.database.data.OrderedColumns;
+import org.opendatakit.database.data.TypedRow;
+import org.opendatakit.database.data.UserTable;
+import org.opendatakit.database.service.DbHandle;
+import org.opendatakit.database.service.UserDbInterface;
+import org.opendatakit.exception.ActionNotAuthorizedException;
+import org.opendatakit.exception.ServicesAvailabilityException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ServerDatabaseCommunicator {
+
+    private  static String appName;
+    private  static UserDbInterface userDbInterface;
+    private static String userId;
+    private static DbHandle dbHandle;
+
+    private static final String USER_TABLE_ID = "UsersTable";
+    private static final String GROUPS_TABLE_ID = "GroupsTable";
+    private static final  String NOTIFICATIONS_TABLE_ID = "NotificationsTable";
+    private static final String RESPONSES_TABLE_ID = "ResponsesTable";
+
+    private static final String USER_ID = "UserId";
+    private static final String USER_NAME = "UserName";
+
+    private static final String GROUP_LIST = "GroupList";
+    private static final String GROUP_NAME = "GroupName";
+    private static final String GROUP_ID = "GroupId";
+
+    private static final String RESPONSE_ID = "ResponseId";
+    private static final String RESPONSE_TEXT = "ResponseText";
+    private static final String RESPONSE_TIME = "ResponseTime";
+
+
+    private static final String NOTIFICATION_ID = "NotificationId";
+    private static final String NOTIFICATION_TITLE = "NotificationTitle";
+    private static final String NOTIFICATION_MESSAGE = "NotificationMessage";
+    private static final String NOTIFICATION_TYPE = "NotificationType";
+    private static final String NOTIFICATION_TIME = "NotificationTime";
+
+    private static final String ANONYMOUS_USER_NAME = "anonymous";
+    private static final String USERNAME_PREFIX = "username:";
+
+    private static final String SPLIT_REGEX_STRING = ",";
+
+
+    private static final List<String> USERS_TABLE_COLUMNS_LIST = Arrays.asList(USER_ID,
+            USER_NAME, GROUP_LIST);
+
+    private static final List<String> RESPONSES_TABLE_COLUMNS_LIST = Arrays.asList(RESPONSE_ID,
+            RESPONSE_TEXT, NOTIFICATION_ID, USER_NAME, RESPONSE_TIME);
+
+    private static ServerDatabaseCommunicator serverDatabaseCommunicator;
+
+    private ServerDatabaseCommunicator() {
+
+    }
+
+    public static synchronized ServerDatabaseCommunicator getInstance() {
+
+        if (serverDatabaseCommunicator == null) {
+            serverDatabaseCommunicator = new ServerDatabaseCommunicator();
+        }
+
+        return serverDatabaseCommunicator;
+    }
+
+    public void init(UserDbInterface userDb, String aName) throws ServicesAvailabilityException, JSONException, ActionNotAuthorizedException {
```

Comment:
The public methods also needs to be made synchronized

Reply:
Done.